### DISCUSSION TITLE
MPlayer Engine Update?

### DISCUSSION THREAD

**Discussion by Raza-0007** [smplayer-dev/smplayer#729] [General]
To smplayer developer,

Is there any chance of updating the MPlayer engine inside smplayer? 

I know its not a priority, as the default engine of smplayer is mpv, but the included MPlayer engine, SVN r38151, is from 08/2019, and will be 4 years old this August. The latest MPlayer build is SVN r38407, released on 12/17/2022. 

See the official MPlayer page "the binaries section" here: http://www.mplayerhq.hu/design7/dload.html
Which leads to this page here: https://oss.netfarm.it/mplayer/
Which leads to this Sourceforge repository here: https://sourceforge.net/projects/mplayer-win32/files/MPlayer%20and%20MEncoder/

I personally don't use MPlayer, but it is nice to have the option to switch to a different engine to see the difference once in a while. 

Also if someone wants to update MPlayer themselves, as long as they are using the portable version of smplayer, all they need to do is to download the latest SVN from the Sourceforge link above, unzip it, and then remove the contents of the mplayer folder inside smplayer portable directory, and replace it with the contents of the unzipped MPlayer folder. Switch multimedia engines inside smplayer and you are good to go. I have tested it on newest development build and everything seem to work fine on default settings.
.
.
.

 
![Mplayer Update](https://github.com/smplayer-dev/smplayer/assets/131807929/fd549ded-6057-4572-bd24-029646f85d0d)

**Comment by Raza-0007** [comment]
Thanks Ricardo,

I noticed that you upgraded the MPlayer engine in the latest development Windows build (2023.06.20) to SVN r38407.

But I also noticed that you downgraded the mpv engine to 0.35 from 0.35.1. Was there a reason for this downgrade? A bug or something in mpv 0.35.1?

Thanks,

~Raza

**Comment by smplayer-dev** [comment]
I took one of the development versions, and that was to fix playback of youtube with no audio when using mpv + yt-dlp.

**Reply by Raza-0007** [reply]
I see. I did not realize that the version "0.35.0-436-g1c82d6ae" (June 10, 2023) is one of the later versions, and came after the official release 0.35.1 (Jan 28, 2023).  

It is strange that mpv devs went back to 0.35.0 numbering after releasing 0.35.1
